src/h2load.cc

```diff
@@ -178,6 +180,41 @@ void second_timeout_w_cb(struct ev_loop *loop, ev_timer *w, int revents) {
 }
 } // namespace
 
+namespace {
+// Called when an a connection has been inactive for a set period of time
+void conn_inactivity_timeout_cb(EV_P_ ev_timer *w, int revents) {
+  auto client = static_cast<Client *>(w->data);
+  ev_timer_stop(client->worker->loop, &client->conn_inactivity_watcher);
+
+  if (client->worker->config->conn_active_timeout > 0 &&
+      ev_is_active(&client->conn_active_watcher)) {
+    ev_timer_stop(client->worker->loop, &client->conn_active_watcher);
+  }
+
+  if (util::check_socket_connected(client->fd)) {
+    client->timeout();
+  }
+}
+} // namespace
+
+namespace {
+// Called a fixed amount of time after all requests have been made on a
+// connection
+void conn_active_timeout_cb(EV_P_ ev_timer *w, int revents) {
+  auto client = static_cast<Client *>(w->data);
+  ev_timer_stop(client->worker->loop, &client->conn_active_watcher);
+
+  if (client->worker->config->conn_inactivity_timeout > 0 &&
+      ev_is_active(&client->conn_inactivity_watcher)) {
+    ev_timer_stop(client->worker->loop, &client->conn_inactivity_watcher);
+  }
+
+  if (util::check_socket_connected(client->fd)) {
+    client->timeout();
+  }
+}
+} // namespace
```

Comment:
These 2 functions are essentially same, except for stopping individual timers.
As I wrote below, if we are going to initialize always these timers, it is safe to just stop both here without checking it is activated.  Then, we can achieve the same result using just one function.